apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-inserter-menu-search-term.js

```diff
@@ -42,11 +43,33 @@ const InserterMenuTrackingEvent = function () {
 		tracksRecordEvent( 'wpcom_block_picker_no_results', eventProperties );
 	}, 500 );
 
+	const debouncedSetFilterValueDiscovery = debounce( ( search_term ) => {
+		if ( search_term.length < 3 ) {
+			return;
+		}
+
+		const blockDirectoryResults = document.querySelectorAll(
+			'.block-directory-downloadable-block-list-item'
+		).length;
+
+		if ( blockDirectoryResults ) {
+			tracksRecordEvent( 'wpcom_block_discovery_results', {
+				search_term,
+				results: blockDirectoryResults,
```

Comment:
Would it make sense to include whether there were already installed blocks or patterns? When there are items already it makes it less likely that the user will actually have seen the block discovery results, so maybe you'd want to be able to filter those out of your analytics queries?

```suggestion
				search_term,
				hasItems,
				results: blockDirectoryResults,
```

Reply:
You're right, I didn't see the discovery results when results appear on top, I will ask if we should filter it!

Reply:
I [asked here](p1666016039170339-slack-CKZHG0QCR) and we will use same logic as `block search` events.